packages/cursorless-engine/src/actions/Remove.ts

```diff
@@ -28,28 +35,45 @@ export default class Delete implements SimpleAction {
     }
 
     const thatTargets = flatten(
-      await runOnTargetsForEachEditor(targets, async (editor, targets) => {
-        const edits = targets.map((target) => target.constructRemovalEdit());
-        const ranges = edits.map((edit) => edit.range);
-
-        const [updatedRanges] = await performEditsAndUpdateRanges(
-          this.rangeUpdater,
-          ide().getEditableTextEditor(editor),
-          edits,
-          [ranges],
-        );
-
-        return zip(targets, updatedRanges).map(
-          ([target, range]) =>
-            new RawSelectionTarget({
-              editor: target!.editor,
-              isReversed: target!.isReversed,
-              contentRange: range!,
-            }),
-        );
-      }),
+      await runOnTargetsForEachEditor(targets, this.runForEditor),
     );
 
     return { thatTargets };
   }
+
+  private async runForEditor(editor: TextEditor, targets: Target[]) {
+    const edits = targets.map((target) => target.constructRemovalEdit());
+
+    const cursorSelections = { selections: editor.selections };
+    const editSelections = {
+      selections: edits.map(
+        ({ range }) => new Selection(range.start, range.end),
+      ),
+      rangeBehavior: RangeExpansionBehavior.closedClosed,
```

Comment:
I don't think behaviour makes a difference for deletions; only insertions. Tho I would assume you switched to this more complex api because something wasn't working? Or maybe you copy-pasted this code from an action that performs insertions? In which case I'd try to switch back to the simpler api

Reply:
The original code was using `performEditsAndUpdateRanges` https://github.com/cursorless-dev/cursorless/blob/main/packages/cursorless-engine/src/actions/Remove.ts#L35C39-L35C66 which calls `rangesToSelectionInfos` https://github.com/cursorless-dev/cursorless/blob/main/packages/cursorless-engine/src/core/updateSelections/updateSelections.ts#L309 which uses `RangeExpansionBehavior.closedClosed` https://github.com/cursorless-dev/cursorless/blob/main/packages/cursorless-engine/src/core/updateSelections/updateSelections.ts#L100 so didn't want to change that behaviour from original code.

OK I guess I could use `performEditsAndUpdateSelections` instead of `performEditsAndUpdateSelectionsWithBehavior`?

Reply:
And yes I had copied it from the "insert" action.

Reply:
Yeah I'd try using `performEditsAndUpdateSelections` just because it's a simpler api